### PR TITLE
Archive and empty data-dir if member dir exists

### PR DESCRIPTION
**Pull request by retroflexer**
If a member dir is present in datadir, archive and empty the dirty datadir.

The recursive copy here assumes a simple directory structure and dealing with regular files. 
This code doesn't take any special care about:
1. Detecting loops in the directory recursion
2. Effort to retain symbolic links
3. Detection of named pipes
4. Special dealing with device files

Please read https://github.com/etcd-io/etcd/blob/master/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by retroflexer** [comment]
This is needed, because rename doesn't work across filesystems.

File: openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go [hunk 1]
```diff
@@ -198,20 +199,83 @@ func (o *DiscoverEtcdInitialClusterOptions) Run() error {
 	return nil
 }
 
-// TO DO: instead of archiving, we should remove the directory to avoid any confusion with the backups.
-func archiveDataDir(sourceDir string) error {
+func archiveAndEmptyDataDir(sourceDir string) error {
```
Comment (deads2k) [inline openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go]: in godoc, explain in three sentences why we do this.

File: openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go [hunk 2]
```diff
@@ -198,20 +199,83 @@ func (o *DiscoverEtcdInitialClusterOptions) Run() error {
...
 		targetDir = filepath.Join(sourceDir+"-removed-archive", time.Now().Add(time.Second).Format(time.RFC3339))
 	}
-	if err := os.Rename(sourceDir, targetDir); err != nil && !os.IsNotExist(err) {
+	if err := CopyAndRemoveDir(sourceDir, targetDir); err != nil {
 		return err
 	}
 	return nil
 }
 
+func CopyAndRemoveDir(source string, dest string) (err error) {
```
Comment (deads2k) [inline openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go]: scope to private

File: openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go [hunk 3]
```diff
@@ -198,20 +199,83 @@ func (o *DiscoverEtcdInitialClusterOptions) Run() error {
...
+				// continue despite errors
+				fmt.Fprintln(os.Stderr, "Failed to copy %s: %v", sourcefilename, err)
+			}
+		}
+		os.RemoveAll(sourcefilename)
+	}
+	return
+}
+
+func CopyFile(source string, dest string) (err error) {
```
Comment (deads2k) [inline openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go]: scope to private

**Comment by retroflexer** [comment]
/hold 

This change is perhaps not needed.

**Comment by retroflexer** [comment]
/hold 

This change is perhaps not needed.

**Pull request closed without merging**